pkg/process/util/network_filter.go

```diff
@@ -0,0 +1,73 @@
+package util
+
+import (
+	"net"
+	"strconv"
+	"strings"
+
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+// ConnectionFilter holds a user-defined blacklisted IP/CIDR, and ports
+type ConnectionFilter struct {
+	IP    *net.IPNet
+	Ports map[uint16]struct{}
+}
+
+// ParseBlacklist takes the user defined blacklist and returns a slice of ConnectionFilters
+func ParseBlacklist(filters map[string][]string) (blacklist []*ConnectionFilter) {
+	for ip, ports := range filters {
+		conn := &ConnectionFilter{Ports: map[uint16]struct{}{}}
+
+		// retrieve valid IPs
+		if strings.Contains(ip, "/") {
+			_, subnet, err := net.ParseCIDR(ip)
+			if err != nil {
+				log.Debugf("Could not parse %s", err)
+			}
+			conn.IP = subnet
+		} else if strings.Contains(ip, ":") {
+			_, subnet, err := net.ParseCIDR(ip + "/64") // if given ipv6, prefix length of 64
+			if err != nil {
+				log.Debugf("Could not parse %s", err)
+			}
+			conn.IP = subnet
+
+		} else {
+			_, subnet, err := net.ParseCIDR(ip + "/32") // if given ipv4, prefix length of 32
+			if err != nil {
+				log.Debugf("Could not parse given IPs: %s", err)
+			}
+			conn.IP = subnet
+		}
+
+		// convert and store given ports
+		for _, v := range ports {
```

Comment:
If we create a filter and successfully parse the CIDR, but not the port, then we're going to skip every port, right?

Perhaps we should be extra safe and skip the whole filter if any parsing fails